doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -650,6 +650,9 @@ This section describes some special purpose tactics to work with
       implicitly added to this table when defined.
       Use the :cmd:`Add` and :cmd:`Remove` commands to update this set manually.
 
+.. tacn:: simple injection {? @induction_arg }
+   :undocumented:
```

Comment:
From Hugo's explanation, you can add a description for this tactic (and make it a variant of `injection`) saying something like: `simple injection` will always add the derived equalities as new premises in the goal instead of putting them in the context, even if the `Structural Injection` flag is on.

Reply:
Here's what I have now.  I also reworded the description of the flag, dropping "erases the original hypothesis" because I expect `injection` always does that; the flag doesn't change that.  Is that correct?

   .. tacn:: simple injection {? @induction_arg }

      Similar to :tacn:`injection`, but always adds the derived equalities
      as new :term:`premises <premise>` in the current goal (instead of as
      new hypotheses) even if the :flag:`Structural Injection` flag is set. 

   .. flag:: Structural Injection

      When set, :n:`injection @term` adds the generated equalities as
      new hypotheses rather than adding them to the current goal as
      :term:`premises <premise>`, as if giving :n:`injection @term as`
      (with an empty list of names). This flag is off by default.

Reply:
Actually no, it only erases the original hypothesis if `as` is used or `Structural Injection` is set...

Reply:
OK, then

When this :term:`flag` is set, :n:`injection @term` erases the original hypothesis
      and adds the generated equalities as new hypotheses rather than adding them
      to the current goal as :term:`premises <premise>`, as if giving :n:`injection @term as`
      (with an empty list of names). This flag is off by default.